Signal/src/ViewControllers/ConversationView/Cells/OWSMessageCell.m

```diff
@@ -1243,6 +1274,30 @@ - (void)setIsCellVisible:(BOOL)isCellVisible {
 
 #pragma mark - Gesture recognizers
 
+- (void)handleTapGesture:(UITapGestureRecognizer *)sender
+{
+    OWSAssert(self.delegate);
+
+    if (sender.state != UIGestureRecognizerStateRecognized) {
+        DDLogVerbose(@"%@ Ignoring tap on message: %@", self.logTag, self.viewItem.interaction.debugDescription);
+        return;
+    }
+
+    if (self.lastBodyMediaView) {
+        // Treat this as a "body media" gesture if:
+        //
+        // * There is a "body media" view.
+        // * The gesture occured within or above the "body media" view.
+        CGPoint location = [sender locationInView:self.lastBodyMediaView];
+        if (location.y <= self.lastBodyMediaView.height) {
```

Comment:
Not a big deal, but this strikes me as more error prone than adding separate explicit gestures to the views, and I think the problem will become worse as we add more views, e.g. if we change margins and don't properly account for it here.

Reply:
I guess its subjective, but...

* It's not as simple as "media gesture" v. "text gesture."  We'll also have gestures that originate in the "status indicator", the "failed to send indicator", the "tap for more view", the margins/whitespace/bubble edges of the cell.  
* IMO we want to be "permissive about what we accept" in terms of gestures". e.g. it would be a bug if taps in the whitespace of the bubble bounds didn't register.
* With quoted replies, we'll soon have more (3 more?) subviews in these cells.

Therefore, I think we'll need to explicitly dispatch these gestures using rules like these.

If we try to resolve these complexities by using more gesture recognizers, we'll leave ourselves vulnerable to difficult bugs: 

* GRs attached to views which overlap.
* Regions of the bubble view not covered by any subview with a GR.
* Visible bounds of contents (e.g. body text) not corresponding exactly to the hot area for that content.  Note: We don't want to try to fix that by ensuring that subviews exactly fill bubble bounds without any overlap; that will make a difficult layout problem very unwieldy.